ibm/service/power/resource_ibm_pi_key.go

```diff
@@ -114,21 +147,70 @@ func resourceIBMPIKeyRead(ctx context.Context, d *schema.ResourceData, meta inte
 	}
 
 	// get key
-	sshkeyC := instance.NewIBMPIKeyClient(ctx, sess, cloudInstanceID)
+	sshkeyC := instance.NewIBMPISSHKeyClient(ctx, sess, cloudInstanceID)
 	sshkeydata, err := sshkeyC.Get(key)
 	if err != nil {
 		return diag.FromErr(err)
 	}
 
-	// set attributes
+	// Arguments
+	d.Set(Arg_CloudInstanceID, cloudInstanceID)
+	d.Set(Arg_Description, sshkeydata.Description)
+	d.Set(Arg_KeyName, sshkeydata.Name)
+	d.Set(Arg_SSHKey, sshkeydata.SSHKey)
+	d.Set(Arg_Visibility, sshkeydata.Visibility)
+
+	// Attributes
 	d.Set(Attr_CreationDate, sshkeydata.CreationDate.String())
 	d.Set(Attr_Key, sshkeydata.SSHKey)
+	d.Set(Attr_SSHKeyID, sshkeydata.ID)
 	d.Set(Attr_Name, sshkeydata.Name)
+	d.Set(Attr_PrimaryWorkspace, sshkeydata.PrimaryWorkspace)
 
 	return nil
 }
 
 func resourceIBMPIKeyUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	// session
```

Comment:
I know we ran into issues in the past with the tests reading too quickly. Was there no issues here without any kind of special wait the update?